README.md

```diff
@@ -66,6 +70,39 @@ A `bfloat16` number is a single-precision float truncated at 16 bits.
 
 Exponent: 8, Mantissa: 7, exponent bias: 127. IEEE 754, with NaN and inf.
 
+### `float4_e2m1`
```

Comment:
Fix names to include `fn` suffix.

I'm actually having trouble finding a great definition of the `f` and `n` suffixes even in the LLVM discussion that added them: I don't suppose you have a link to the definition?

In particular, I'm not sure if `n` should appear in the name, given that `n` also appears in the suffix of FP8 types with a single NaN, but these have no NaN. So I'm a bit unclear what the suffix means.

Reply:
`f` means "finite", `n` means "special NaN representation" (e.g. non-IEEE)
I saw this somewhere in the comments, will post a link once I find it.

Reply:
Fixed the type name.

Reply:
Oh, actually, it's in this same file, below:

`F` is for "finite" (no infinities), `N` for with special NaN encoding, `UZ` for unsigned zero.

Reply:
I guess one could say that "no NaN encoding" is a "special NaN encoding".

Also, LLVM APFloat.cpp has these types with "FN" suffix:
https://github.com/llvm/llvm-project/blob/5537ae87b3a87b3abeb4e6983cecd9b103648243/llvm/lib/Support/APFloat.cpp#L150

We could probably change the suffix, but we need to be consistent across the repositories.

Reply:
We should agree with LLVM, so that works for me.

Reply:
Not sure if you didn't push the fix yet, the headers are still suffix-less.

Reply:
Fixed now.